### PR TITLE
[ParseableInterface] Turn on -enable-parseable-module-interface always

### PR DESCRIPTION
**Pull request by jrose-apple**
...and remove the option. This is \~technically\~ CLI-breaking because Swift 5 shipped this as a hidden driver option, but it wouldn't have *done* anything in Swift 5, so I think it's okay to remove.

Note that if a parseable interface (.swiftinterface) and a binary interface (.swiftmodule) are both present, the binary one will still be preferred. This just *allows* parseable interfaces to be used.

This is made possible by @harlanhaskins and @nathawes' work over the last two months! There's still more to be done, but we're getting to a livable state, and it's great.

rdar://problem/36885834

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Comment by jrose-apple** [comment]
@swift-ci Please test compiler performance

File: lib/Frontend/ParseableInterfaceModuleLoader.cpp [hunk 1]
```diff
@@ -713,9 +712,6 @@ class ParseableInterfaceModuleLoaderImpl {
     if (validationInfo.status != serialization::Status::Valid)
       return false;
 
-    assert(validationInfo.name == moduleName &&
```
Comment (harlanhaskins) [inline lib/Frontend/ParseableInterfaceModuleLoader.cpp]: Any reason this was removed as part of these changes? I think it's fine to remove.

**Review by harlanhaskins — APPROVED** [review]
:tada:

**Review by nathawes — APPROVED** [review]

File: lib/Frontend/ParseableInterfaceModuleLoader.cpp (hunk 1 above)
Comment (jrose-apple) [reply lib/Frontend/ParseableInterfaceModuleLoader.cpp]: The change uncovered a bug: now that we use this function for adjacent swiftmodules as well as cached swiftmodules, the assertion is no longer valid. This was tested by test/Serialization/load-wrong-name.swift.

**Comment by jrose-apple** [comment]
Looks like this may need a paired LLDB change too.

**Comment by jrose-apple** [comment]
apple/swift-lldb#1386
@swift-ci Please test

**Comment by jrose-apple** [comment]
\*sigh\* Those improvements are not possible, so something must still be wrong with this bot. Merging.

**Pull request merged**